### ISSUE TITLE
Cannot read property 'avatar' of null 

### ISSUE THREAD

**Issue by pavelbinar** [jlesage/docker-nginx-proxy-manager#21]
I sotopped seeing the Proxy Hosts settings and instead I got the:

> Cannot read property 'avatar' of null 

error message. I do not know how that happened, but the first guess is the automatics docker image update `jlesage/nginx-proxy-manager`

I am using `v2.0.13`

**Comment by jlesage** [comment]
Where do you get this error exactly?  In the UI or the container's log?  Is the UI still accessible?

**Comment by pavelbinar** [comment]
The error is in the GUI:

![Screenshot 2019-06-04 13 50 37](https://user-images.githubusercontent.com/1060032/58876957-d2882200-86cf-11e9-984b-614dfd616ccb.png)
![Screenshot 2019-06-04 13 49 58](https://user-images.githubusercontent.com/1060032/58876958-d2882200-86cf-11e9-8988-f93833ebcf04.png)

**Comment by jlesage** [comment]
How did you change your avatar? I think this is not something that can be changed via the interface.

**Comment by pavelbinar** [comment]
I did not change it nor set it up :)

I guess it uses https://en.gravatar.com/

**Comment by jlesage** [comment]
Yes you are right...

So what this command outputs?

```
docker exec <conainer name> sh -c 'echo "select email,avatar from nginxproxymanager.user" | mysql'
```

Do you have a null avatar?

**Comment by pavelbinar** [comment]
Here is the result of your query:

```
root@Xenon:~# docker exec NginxProxyManager sh -c 'echo "select email,avatar from nginxproxymanager.user" | mysql'
email	avatar
pavel@XXXX	//www.gravatar.com/avatar/13711f8a5ed4b796fe96cd42e26aca90?default=mm
pavel@XXXX	//www.gravatar.com/avatar/13711f8a5ed4b796fe96cd42e26aca90?default=mm
```

I modified the email to `pavel@XXXX` just to anonymize the data.

**Comment by pavelbinar** [comment]
Hi @jlesage ,

I am keeping the container with all my issues running so you can debug it.
It do not see any progress so I will delete it and start over.

Let me know if I should wait.

Thanks.

**Comment by jlesage** [comment]
Sorry I didn't have time to work on this.
If it's possible for you to remove any sensitive data from your config and share the appdata folder, I could use it to debug on my side.  Else just start over to have something functional.

**Comment by TWhidden** [comment]
Ran into the same issue here.

**Comment by TWhidden** [comment]
![image](https://user-images.githubusercontent.com/9921122/87979094-5fce3780-ca86-11ea-8cdd-272db668709e.png)

**Comment by DragonicDefson** [comment]
This also happens when you delete an user account.

**Comment by kevinsherman** [comment]
I had this problem when re-creating the user account due to a lost password. I had to update a few tables (proxy_host, redirection_host, audit, etc.) Once those were referencing the correct owner_user_id, the error went away (and I was able to see/edit my proxy entries).

**Comment by bugs181** [comment]
> I had this problem when re-creating the user account due to a lost password. I had to update a few tables (proxy_host, redirection_host, audit, etc.) Once those were referencing the correct owner_user_id, the error went away (and I was able to see/edit my proxy entries).

This is the error I'm having. Tried unsuccessfully resetting my password following the directions from #230 . Now I'm receiving avatar errors like OP. Could you elaborate on the commands to update the tables? I'm not familiar with MariaDB / MySQL at all.

**Comment by xcad2k** [comment]
> > I had this problem when re-creating the user account due to a lost password. I had to update a few tables (proxy_host, redirection_host, audit, etc.) Once those were referencing the correct owner_user_id, the error went away (and I was able to see/edit my proxy entries).
> 
> This is the error I'm having. Tried resetting my password following the directions from [#230](https://github.com/jc21/nginx-proxy-manager/issues/230). Now I'm receiving avatar errors like OP. Could you elaborate on the commands to update the tables? I'm not familiar with MariaDB / MySQL at all.
> 
> UPDATE:
> After replicating the steps from [#230](https://github.com/jc21/nginx-proxy-manager/issues/230) again, I was able to get it to a relatively stable state. It appears that I'm going to be unable to delete old accounts until this is fixed. My last admin account was using an `@icloud.com` suffix which could explain the avatar issue.
> Setting all of the users to `is_deleted=0;` via `UPDATE user SET is_deleted=0;` will make them re-appear in the Users list and I'm able to access the Hosts list once again.
> 
> Output of:
> `select email,avatar from user;`
> 
> ```
> +--------------------+-----------------------------------------------------------------------+
> | email              | avatar                                                                |
> +--------------------+-----------------------------------------------------------------------+
> | myuser@icloud.com | //www.gravatar.com/avatar/c203a2090b903962ae58d5fb3d3XXXXX?default=mm |
> | myuser@icloud.com | //www.gravatar.com/avatar/c203a2090b903962ae58d5fb3d3XXXXX?default=mm |
> | myuser@gmail.com  | //www.gravatar.com/avatar/02d753f45ef41e670c41a6a81c8XXXXX?default=mm |
> | myuser@icloud.com | //www.gravatar.com/avatar/c203a2090b903962ae58d5fb3d3XXXXX?default=mm |
> | myuser@icloud.com | //www.gravatar.com/avatar/c203a2090b903962ae58d5fb3d3XXXXX?default=mm |
> | admin@example2.com | //www.gravatar.com/avatar/6257524feec9cca05f182d70169XXXXX?default=mm |
> +--------------------+-----------------------------------------------------------------------+
> ```

This also worked for me! Thanks

**Comment by iamenderst** [comment]
Just ran into this after deleting a user. Have to change owner_user_id in the tables of the affected pages to an existing user. For me was access_list and proxy_host tables.

**Comment by zygimantus** [comment]
Yes, just follow this guide and you won't have any issues https://github.com/NginxProxyManager/nginx-proxy-manager/issues/230#issuecomment-815078355

**Comment by mcarbonneaux** [comment]
in fact the back rest (/api/nginx/certificates?expand=owner,  api return `null` on `owner` attribute if the owner of the proxy host/certificats/etc object is disabled of removed (remove by the ui)...

i think is bug on the api, i think the remove of user must reassign the object to the user who remove user.

and if the user is disabled the api must return the user information and not null.

**Comment by hend0012** [comment]
Hi,

Just delete all users you don't need, then for the user you want do, UPDATE user SET id=1 WHERE id=(Current id of your User);

The issue is, that your main user with id 1 is most likely the owner and the user you want to reactivate. But because your new user does not have id 1 it will give you this error. You could update the id for all tables/entries manually. But as I'm lazy, and only have one user this is a lot quicker.

**Comment by omuleanu** [comment]
I got this problem after resetting my password as shown here: 
https://github.com/NginxProxyManager/nginx-proxy-manager/discussions/1634

And was able to fix it only by deleting all the contents of the data folder (docker volume)

**Comment by naitbrahim** [comment]
If you were playing with adding users , I had an error in the  audit tab, I fixed by "delete from audit_log;" now all the tabs are working fine.
and this is just FYI all the tables using the column "owner_user_id" :
- proxy_host
- certificate
- access_list
- dead_host
- redirection_host
- stream

**Comment by oswaldom-code** [comment]
For my part, just like you, I lost the administrator user's password and did the same as you did, I forced the creation of an administrator user by running the query 'UPDATE user SET is_deleted=1'. And once I logged in with the default credentials, I changed the previous administrator user's password. And that's it, it worked well for me.

**Comment by ZeusJCG** [comment]
> Tuve este problema al volver a crear la cuenta de usuario debido a una contraseña perdida. Tuve que actualizar algunas tablas (proxy_host, redirection_host, auditoría, etc.) Una vez que hacían referencia al propietario_user_id correcto, el error desapareció (y pude ver/editar mis entradas proxy).

Aquí la respuesta: https://www.kalecreates.com/Nginxproxymanager%20Issue%20Fix.html
Yo me traje el archivo .sqlite y lo modifique desde local
Hay que buscar el id del usuario eliminado en las tablas y reemplazarlo por uno que si exista
Después hay que regresar el archivo a su origen y podrán visualizar el panel sin problema
